hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -235,6 +235,13 @@ public class HoodieTableConfig extends HoodieConfig {
       .withDocumentation("Comma-separated list of metadata partitions that have been completely built and in-sync with data table. "
           + "These partitions are ready for use by the readers");
 
+  public static final ConfigProperty<Boolean> APPEND_ONLY_TABLE = ConfigProperty
+      .key("hoodie.table.append.only")
```

Comment:
Boolean flags are really hard to evolve and better reserved for toggle-like configs (switching on/off).

In that case if we'd consider adding "table-type" we'd be exposed to can of warms of banning various permutations of this configs.

I'd suggest to generalize this to be a "table-type" config which we can declare either "mutable" or "immutable"